pkg/services/hcloud/server/server.go

```diff
@@ -385,9 +385,9 @@ func (s *Service) handleBootStateEnablingRescue(ctx context.Context, server *hcl
 			"finishedSince", time.Since(action.Finished),
 			"actionStatus", action.Status)
 
-		hm.Status.ActionIDEnableRescueSystem = actionDone
+		hm.Status.ExternalIDs.ActionIDEnableRescueSystem = actionDone
 		conditions.MarkFalse(hm, infrav1.ServerAvailableCondition,
```

Comment:
I think this is also a place rather for an event than a condition, if we want anything